osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/securitygroup/MgrSecurityGroupCheckMetaTask.java

```diff
@@ -30,168 +30,170 @@
+
+    @Override
+    public void executeTransaction(EntityManager em) throws Exception {
+        this.vs = em.find(VirtualSystem.class, this.vs.getId());
+
+        if (this.vs.getMgrId() == null) {
+            log.warn("Manager VSS Device is not yet present. Do nothing.");
+            return;
+        }
+
+        this.tg = new TaskGraph();
+        ManagerSecurityGroupApi mgrSgApi = this.apiFactoryService.createManagerSecurityGroupApi(this.vs);
+
+        if (this.sg == null) {
+            // Sync all security groups across the vs'es
+            List<? extends ManagerSecurityGroupElement> mgrEndpointGroups;
+            try {
+                mgrEndpointGroups = mgrSgApi.getSecurityGroupList();
+            } finally {
+                mgrSgApi.close();
+            }
+
+            for (SecurityGroupInterface sgi : this.vs.getSecurityGroupInterfaces()) {
+                SecurityGroup sg = sgi.getSecurityGroup();
+                ManagerSecurityGroupElement mepg = findByMgrSecurityGroupId(mgrEndpointGroups,
+                        sgi.getMgrSecurityGroupId());
+                if (mepg == null) {
+                    // It is possible it exists but have not been persisted in database.
+                    // Search security group by name
+                    mepg = findBySecurityGroupByName(mgrEndpointGroups, sg.getName());
+                }
+                if (!sgi.getMarkedForDeletion()) {
+                    if (mepg == null) {
+                        // Add new security group to Manager
+                        this.tg.appendTask(this.createMgrSecurityGroupTask.create(this.vs, sg, sgi));
+                    } else {
+                        this.tg.appendTask(this.updateMgrSecurityGroupTask.create(this.vs, sg, sgi));
+                    }
+                } else if (mepg != null) {
+                    this.tg.appendTask(this.deleteMgrSecurityGroupTask.create(this.vs, mepg));
+                }
+            }
+
+            // Remove any security groups which has no policy binding
+            for (ManagerSecurityGroupElement mepge : mgrEndpointGroups) {
+                // Check if SG present in our Database
+                SecurityGroup sg = findVmidcSecurityGroupByMgrId(em, mepge);
+                if (sg == null || sg.getSecurityGroupInterfaces().isEmpty()) {
+                    // Delete security group from Manager if SG is not in our DB
+                    this.tg.appendTask(this.deleteMgrSecurityGroupTask.create(this.vs, mepge));
+                }
+            }
+        } else {
+            // Sync members only for the specified security group
+            // For a given VS and SG, we will only have one SGI
+            SecurityGroupInterface sgi = SecurityGroupInterfaceEntityMgr
+                    .findSecurityGroupInterfacesByVsAndSecurityGroup(em, this.vs, this.sg);
+            ManagerSecurityGroupElement mepg = mgrSgApi.getSecurityGroupById(sgi.getMgrSecurityGroupId());
+            if (mepg == null) {
```

Comment:
you should add the same check here. Only if SGI is not marked for deletion the create/update actions are valid right?